types/wordpress__customize/index.d.ts

```diff
@@ -0,0 +1,80 @@
+// Type definitions for non-npm package @wordpress/customize-browser 5.5
```

Comment:
Not sure if using a scoped package for this is appropriate (and if it is, should the folder should be `wordpress__customize-browser`?).

I hate to bother you @whyisjake, but are there any plans to publish [customize](https://core.trac.wordpress.org/browser/trunk/src/js/_enqueues/wp/customize/) on npm?

Reply:
Yeah, the folder probably should be renamed...

Not a WP dev, but I don't see this API being published on NPM...

Reply:
Not sure I am the right person to ping here.

Reply:
I'm not an authority, but I don't know of any plans to package and publish customize on npm and would be surprised if that happened.

Is there any precedent for putting unpublished packages in DefinitelyTyped?

As I understand, DT exists to provide types for _published packages_ that don't include types.

Older WordPress script dependencies like customize are in an interesting place. They're not imported via JavaScript but must be enqueued through a separate mechanism and expose functionality in the global space. They're disconnected from any module system and I'm not aware of any plans to reconcile this. I mention this because it makes it difficult to suggest an alternative for where to provide these types 😕 

- The core WordPress project might accept them, but that's unprecedented.
- These older WordPress scripts might find a home alongside other newer scripts in [the Gutenberg project](https://github.com/WordPress/gutenberg/tree/master/packages) and in the `@wordpress` organization on npm. That is likely a significant amount of work for limited benefit. This might be nice as an intermediate step where placeholder packages could be imported via the JavaScript module system to connect modern scripts to older WordPress scripts. This would be a nice home for the types, although this would need to be proposed to the WordPress project.
- The quickest and shortest way to handle this would be to just publish these independently on npm.

I understand the motivation and appreciate the effort in providing these types, but they seem to be an odd fit for DefinitelyTyped. I haven't done a lot of work with DT, so I may simply be ignorant 🙂

Reply:
Hmm, I'm not a DT or a WP dev either. Either way, I'm open to submitting the typings elsewhere if that's a better solution...

I don't really see the API being moved to Gutenberg without a major rewrite (the API's a bit antiquated in design and would need a lot of work I think).

Thanks